### ISSUE TITLE
Bad discount display when specific price is configured with new product price and discount in FO

### ISSUE THREAD

**Issue by Robin-Fischer-PS** [PrestaShop/PrestaShop#15866]
**Describe the bug**
When quantity discount are combined (fixed price + discount), and when FO display is in taxe incl. or excl., the "Discount" display on the product page is false, in 3 cases : 

* When Discount is set in tax excl. and FO Display is set in tax excl. (Case 1)
* When Discount is set in tax incl. and FO Display is set in tax excl. (Case 2)
* When Discount is set in tax excl. and FO Display is set in tax incl. (Case 3)

It is OK in 1 case :

* When Discount is set in tax incl. and FO Display is set in tax incl. (Case 4)

**To reproduce :**

Taxes are enabled in International > Taxes
Display discounted price is set to No in Shop parameters > Products
Taxes display is set to "excl." or "incl." in Shop parameters > Customer Settings > Groups > Edit a group
Quantity discounts is set in Catalog > Product > Pricing > Add a specific price

**FO Display : Tax excl.**

```
• Case 1 « Fixed product price tax excl. + Discount tax excl. » :
    ◦ Product price : 10€ tax excl.
    ◦ VAT 20 %
    ◦ Set specific price for 10 units
    ◦ Fixed product price : 9€ (tax excl.)
    ◦ Discount : -2€ (tax excl.)
    ◦ FO « Discount » Display :
        ▪️ Should be: 3€ (2€ Discount + 1€ fixed price)
        ▪️ Result: NOK, display 2,67€
```

![screenshot display discount - price TE discount TE](https://user-images.githubusercontent.com/55688534/66381757-31c9ac00-e9ba-11e9-8dc7-8a1180352c44.png)
![screenshot display discount - price TE discount TE Display TE](https://user-images.githubusercontent.com/55688534/66381767-38f0ba00-e9ba-11e9-9c09-961225871144.png)

```
• Case 2 « Fixed product price tax excl. + Discount tax incl. » :
    ◦ Product price : 10€ tax excl.
    ◦ VAT 20 %
    ◦ Set specific price for 10 units
    ◦ Fixed product price : 9€ (tax excl.)
    ◦ Discount: -2€ (tax incl.)
    ◦ FO « Discount » Display :
        ▪️ Should be: 2,67€ (1,67€ Discount + 1€ fixed price)
        ▪️ Result: NOK, display 3€
```

![screenshot display discount - price TE discount TI](https://user-images.githubusercontent.com/55688534/66381779-3f7f3180-e9ba-11e9-85d7-7c6882d5b4dc.png)
![screenshot display discount - price TE discount TI Display TE](https://user-images.githubusercontent.com/55688534/66381782-4312b880-e9ba-11e9-9b3e-1f10e5eeeb74.png)

**FO Display : Tax incl.**

```
• Case 3 « Fixed product price tax excl. + Discount tax excl. » :
    ◦ Product price : 10€ tax excl.
    ◦ VAT 20 %
    ◦ Set specific price for 10 units
    ◦ Fixed product price : 9€ (tax excl.)
    ◦ Discount : -2€ (tax excl.)
    ◦ FO « Discount » Display :
        ▪️ Should be: 3,60€ (2,40€ display + 1,20€ fixed price)
        ▪️ Result: NOK, display 4,67€
```

![screenshot display discount - price TE discount TE](https://user-images.githubusercontent.com/55688534/66381798-4b6af380-e9ba-11e9-95b1-ca52eafb34cd.png)
![screenshot display discount - price TE discount TE Display TI](https://user-images.githubusercontent.com/55688534/66381808-4efe7a80-e9ba-11e9-83f1-c9e9a430082f.png)

```
• Case 4 « Fixed product price tax excl. + Discount tax incl. » :
    ◦ Product price : 10€ tax excl.
    ◦ VAT 20 %
    ◦ Set specific price for 10 units
    ◦ Fixed product price : 9€ (tax excl.)
    ◦ Discount: -2€ (tax incl.)
    ◦ FO « Discount » Display :
        ▪️ Should be: 3,20€ (2€ display + 1,20€ fixed price)
        ▪️ Result: OK, display 3,20€
```

![screenshot display discount - price TE discount TI](https://user-images.githubusercontent.com/55688534/66381834-57ef4c00-e9ba-11e9-9ef0-51dacae4875e.png)
![screenshot display discount - price TE discount TI Display TI OK](https://user-images.githubusercontent.com/55688534/66381840-5aea3c80-e9ba-11e9-906a-c2cef1032b31.png)

**Additional information**
PrestaShop version: 1.7.7
PHP version: N/A

**Comment by khouloudbelguith** [comment]
Hi,

`Case1`, the same with PS1.7.6.1.
With PS1.7.5.2, same error
![image](https://user-images.githubusercontent.com/16067358/66412865-9dca0580-e9f6-11e9-968f-9e18be324f35.png)


`Case2`, different result in the FO, but is not correct (1.7.5.2 & PS1.7.6.1)
BO
![image](https://user-images.githubusercontent.com/16067358/66411407-006dd200-e9f4-11e9-9699-5f7ca636f8a0.png)
1.7.6.1 FO
![image](https://user-images.githubusercontent.com/16067358/66411457-0fed1b00-e9f4-11e9-944a-f09488d90993.png)
1.7.5.2 FO
![image](https://user-images.githubusercontent.com/16067358/66412750-6a877680-e9f6-11e9-83ac-bd15226a7e1a.png)

`Case3`, Same with PS1.7.6.1 & PS1.7.5.2

`Case4`, it is Ok with PS1.7.6.1 & PS1.7.5.2 also


Thanks!

**Comment by levyn** [comment]
The current problem has been solved here - #16138

**Comment by sector7digital** [comment]
quick question: does that now mean I have to wait for the 1.7.9.0 version and then it will be fixed? any eta? :)

**Comment by kpodemski** [comment]
@oliiix 

You don't need to wait. The fix is already available in PrestaShop 8.0 (There won't be 1.7.9), **and** you can apply the fix directly to your store:
https://github.com/PrestaShop/PrestaShop/pull/16138/files or hire someone to do that for you :)
